src/app/activity/bounty/Claim.ts

```diff
@@ -68,56 +100,147 @@ const getDbHandler = async (request: ClaimRequest): Promise<{dbBountyResult: Bou
     }
 }
 
-const writeDbHandler = async (request: ClaimRequest, claimedByUser: GuildMember): Promise<void> => {
+const writeDbHandler = async (request: ClaimRequest, dbBountyResult: BountyCollection, claimedByUser: GuildMember): Promise<BountyCollection> => {
     const db: Db = await MongoDbUtils.connect('bountyboard');
-	const bountyCollection = db.collection('bounties');
+    const bountyCollection = db.collection('bounties');
+    let claimedBounty: BountyCollection;
+    const currentDate = (new Date()).toISOString();
+
+    // If claiming an evergreen bounty, create a copy and use that
+    if (dbBountyResult.evergreen) {
+        const bountyRec: BountyCollection = Object.assign({}, dbBountyResult);
+        bountyRec.parentId = bountyRec._id;
+        delete bountyRec._id;
+        delete bountyRec.isParent;
+        delete bountyRec.childrenIds;
+        delete bountyRec.claimLimit;
+        const claimedInsertResult = await bountyCollection.insertOne(bountyRec);
+        if (claimedInsertResult == null) {
+            Log.error('failed to create claimed bounty from evergreen');
+            throw new Error('Sorry something is not working, our devs are looking into it.');
+        }
+        claimedBounty = await bountyCollection.findOne({_id: claimedInsertResult.insertedId});
+        let updatedBountyResult: UpdateWriteOpResult = await bountyCollection.updateOne({ _id: new mongo.ObjectId(dbBountyResult._id) }, {
```

Comment:
Can we use a similar pattern here for the `updateOne` call as we do in other activities? 

`const dbBountyResult: BountyCollection = await bountyCollection.findOne({
		_id: new mongo.ObjectId(request.bountyId),
		status: BountyStatus.in_review,
	});
	const writeResult: UpdateWriteOpResult = await bountyCollection.updateOne(dbBountyResult, {
		$set: {`

Reply:
Changed name. I ran into trouble putting the entire document in as a filter, that's why I limited it to just the _id. If you change anything on that document it can cause unexpected results. For example, if it happens to match multiple documents, it will just silently update the first one it finds. Just using the _id will also catch errors like a document missing an _id.